task_processing/plugins/kubernetes/task_config.py

```diff
@@ -0,0 +1,43 @@
+import re
+
+from pyrsistent import field
+from pyrsistent import PMap
+from pyrsistent import PVector
+
+from task_processing.interfaces.task_executor import DefaultTaskConfigInterface
+
+
+class KubernetesTaskConfig(DefaultTaskConfigInterface):
+    node_selector = field(type=PMap)
+    containers = field(type=PVector)
+    # Hardcoded for the time being
+    restart_policy = "Never"
+    # By default, the retrying executor retries 3 times. This task option
+    # overrides the executor setting.
+    retries = field(type=int,
+                    factory=int,
+                    mandatory=False,
+                    invariant=lambda r: (r >= 0, 'retries >= 0'))
+    volumes = field(type=PVector)
+
+    @property
+    def pod_name(self) -> str:
+        return self.name
+
+    def set_pod_name(self, pod_name: str):
+        try:
+            name, uuid = pod_name.rsplit('.', maxsplit=1)
+        except ValueError:
+            raise ValueError(f'Invalid format for pod_name {pod_name}')
```

Comment:
we'll need to figure out what to do with the uuid (or just ignore it) at some point, but we can probably leave this for now until we're actually working on handling tron restarts (since right now it's not particularly clear that you need to have saved the previous uuid if you want to reconstruct a task config and make it match exactly to the previous one)

Reply:
the point of uuid is to have unique task name, in mesos plugin it's generated automatically upon creating task object: https://github.com/Yelp/task_processing/blob/master/task_processing/plugins/mesos/task_config.py#L60, in kubernetes we probably want to do the same as pod names should also be unique.

Reply:
yea - I was thinking that we could do whatever kubernetes does internally to prevent duplicates (since I remember reading something about how k8s handles conflicts in the docs for a V1PodSpec) - but I guess that with 253 characters we can probably still use a uuid and save ourselves the trouble :)

Reply:
as i recall, k8s itself won't do anything about pod name itself, it'll just error out if you try creating two pods with the same name in the same namespace.

uuids are probably an overkill, i would probably go for a random 6-byte base58 string these days.

we could try to be smart and recognize that those pods will be unique if you use retry number in pod_name, but then retries should be baked into task executor, while right now we have a pretty abstract implementation (even if it's in mesos folder): https://github.com/Yelp/task_processing/blob/cf843658d2af1122e7aeb279200a75aa60c785f5/task_processing/plugins/mesos/retrying_executor.py

Reply:
oh, I was thinking of `generate_name` from https://github.com/kubernetes-client/python/blob/master/kubernetes/docs/V1ObjectMeta.md, but that won't work for us - a random 6-byte base58 string sounds good to me too, I was originally worried about the uuid length :p

Reply:
I was looking for the paasta way of randomly generating the uuid in baseball 58, but could only find encoding in base64. So I just added it manually as there are no python libs for randomly generating base58 strings.